dradis/dradis-ce
# Paginate api

### Summary

Currently, the API endpoints don’t paginate their data. We need the pagination behavior for the DataTables server-side fetching.

**Proposed solution**

Add an optional param: `page` that fetches 25 entries at a time.

### How to test

1. Create a project with a bunch of data.
2. Take note of a user’s api token.
3. Run the following command:

```
curl -k -H 'Authorization: Token token=<token>\n' http://localhost:3000/api/issues\?page\=1
```

4. Confirm that you get only upto 25 entries.
5. Run the command again but this time with `page=2`
6. Test the command in different endpoints: nodes, notes, evidence.


### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!


### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

Please review the [CONTRIBUTING.md](https://github.com/dradis/dradis-ce/blob/master/CONTRIBUTING.md)
file for the details.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List
